### PR TITLE
Update Xib2Nib logic removing UIRoundedRectButton as a separate class

### LINKED ISSUE #1975: Demo app is failing in UIKit!constructObject
Callstack:
00 ucrtbased!common_assert_to_message_box<wchar_t>
01 ucrtbased!common_assert<wchar_t>
02 ucrtbased!_wassert
03 UIKit!constructObject
04 UIKit!idForItem
05 UIKit!getObjectForKey
06 UIKit!-[UINibUnarchiver decodeObjectForKey:]
07 UIKit!-[UIRuntimeEventConnection initWithCoder:]
08 UIKit!constructObject
09 UIKit!idForItem
0a UIKit!constructObject
0b UIKit!idForItem
0c UIKit!getObjectForKey
0d UIKit!-[UINibUnarchiver decodeObjectForKey:]
0e UIKit!-[UINib instantiateWithOwner:options:]
0f UIKit!-[UIViewController loadView]
10 UIKit!-[UIViewController view]
11 UIKit!-[UIWindow setRootViewController:]
12 UIKit!UIApplicationMainInit
13 UIKit!ApplicationMainStart
14 UIKit!RunApplicationMain
15 UIKit!DoApplicationLaunch
16 UIKit!_ApplicationLaunch
17 UIKit!UIApplicationLaunched

### PR DESCRIPTION
**Pull request by oliversa-msft**
We removed the subclassed UIRoundedRectButton from UIButton.mm in commit (87dd17bade). As a result, we need to remove the logic from Xib2Nib that relied on this class implementation

Fixes #1975

### CODE REVIEW

File: tools/vsimporter/xib2nib/UIButton.cpp [hunk 1]
```diff
@@ -253,22 +245,15 @@ void UIButton::InitFromStory(XIBObject* obj) {
     if (type) {
         if (strcmp(type, "roundedRect") == 0) {
             getAttrAndHandle("buttonType");
+             // TODO: Investigate why this value deviates from UIButtonType
```
Comment (jaredhms) [inline tools/vsimporter/xib2nib/UIButton.cpp]: >  // TODO: Investigate why this value deviates from UIButtonType [](start = 11, length = 64)

is it trying to map to system?  seems like an odd place to do it...

Comment (jaredhms) [inline tools/vsimporter/xib2nib/UIButton.cpp]: i really hate all of the magic constants in this codebase :'(

---
In reply to: [102358850](https://github.com/Microsoft/WinObjC/pull/2018#discussion_r102358850) [](ancestors = 102358850)

**Comment by jaredhms** [comment]
:shipit:

File: tools/vsimporter/xib2nib/UIButton.cpp (hunk 1 above)
Comment (oliversa-msft) [inline tools/vsimporter/xib2nib/UIButton.cpp]: We'd have to crack open the NIB from iOS and inspect its values for various UIButton types to see if they align with UIButtonType*. Right now, I don't know if this value is correct - might be a legacy value or just a hack.

---
In reply to: [102358917](https://github.com/Microsoft/WinObjC/pull/2018#discussion_r102358917) [](ancestors = 102358917,102358850)

**Pull request merged**